### PR TITLE
Support limiting location listing depth.

### PR DESCRIPTION
**Pull request by jgadling**
### Summary:
- **What:** Support limiting the listing depth for locations to only division/country/region levels
- **Ticket:** [sc229000](https://app.shortcut.com/genepi/story/229000)
- **Env:** `<rdev link>`

### Notes:
This also includes a fix for our tree build workflow!

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [x] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/aspen/workflows/nextstrain_run/run_nextstrain_mpx.sh [hunk 1]
```diff
@@ -38,6 +38,8 @@ s3_prefix="s3://${aspen_s3_db_bucket}/${key_prefix}"
 # We use a file to pass from `export.py` to `save.py` before writing them to DB
 RESOLVED_TEMPLATE_ARGS_SAVEFILE=/tmp/resolved_template_args.json
 
+mpox_git_rev=$(cd /ncov && git rev-parse HEAD)
```
Comment (danrlu) [inline src/backend/aspen/workflows/nextstrain_run/run_nextstrain_mpx.sh]: should this be `cd /mpox`?

**Review by vincent-czi — APPROVED** [review]
Looks great!

File: src/backend/aspen/api/views/locations.py [hunk 2]
```diff
@@ -30,10 +39,20 @@ async def list_locations(
     db: AsyncSession = Depends(get_db),
     settings: APISettings = Depends(get_settings),
     user: User = Depends(get_auth_user),
+    max_location_depth: LocationDepthEnum = Query(default=LocationDepthEnum.LOCATION),
 ) -> LocationListResponse:
 
     # load the locations.
     all_locations_query = sa.select(Location)  # type: ignore
+    # Find out which columns need to be forced to null based on our max location depth
```
Comment (phoenixAja) [inline src/backend/aspen/api/views/locations.py:47]: would be nice to leave a comment here explaining more why we're doing this?

**Review by phoenixAja — APPROVED** [review]

**Review by danrlu — APPROVED** [review]
Exciting!!

File: src/backend/aspen/workflows/nextstrain_run/run_nextstrain_mpx.sh (hunk 1 above)
Comment (jgadling) [reply src/backend/aspen/workflows/nextstrain_run/run_nextstrain_mpx.sh]: thanks for the catch!

**Pull request merged**